runtime/src/logger.rs

```diff
@@ -13,11 +16,15 @@ use tracing_subscriber::Layer;
 
 pub struct Logger {
     tx: UnboundedSender<LogItem>,
+    state: Arc<Mutex<State>>,
```

Comment:
I thought the deployment state is determined based on the instrumentation happening in the run queue + the deployer tracing layer. Is this something that plays into how the deployment state is computed?

Reply:
In the deployer it is, but we were incorrectly defaulting to `State::Running` for logs received from the runtime here, in try_from: https://github.com/shuttle-hq/shuttle/blob/main/deployer/src/runtime_manager.rs#L131. 

So with this change we rather default to `State::Loading` for runtime logs, and then change it to `State::Running` after the "start" event. There may well be a more graceful way to do this though, I'll try to think about it more!

Reply:
I think this looks good though.

What I missed is that we're subscribing to the runtime logs in the `get_runtime_client`. All runtime logs emitted during the `load` and `run` phases are sent to persistence, but just before that, the deployment state is updated according to the log state field.

Also, the runtime logger gets instantiated in the `load` phase but is moved alongside the `loader` in a task that outlives the `load` handler scope, and even the `load` phase, being registered as a tracing subscriber. This means all the runtime logs (including the `start` phase ones) are sent on the channel created by the `SubscribeLogs` RPC, called in the `get_runtime_client`, so eventually the runtime's log state will be sent to the deployer, and the deployment state updated.